src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,211 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import geometry.euclidean.basic
+import geometry.manifold.charted_space
+import analysis.normed_space.inner_product
+
+/-!
+# Conformal Maps on Inner Product Spaces
+
+A map between real inner product spaces `X`, `Y` is `conformal_at` some point `x` in `X` if it is
+real differentiable and its differential at that point is a nonzero constant multiple of
+a linear isometry equivalence.
+
+## Main results
+
+* `conformal_at`: the main definition
+* The conformality of the composition of two conformal maps, the identity map
+  and multiplications by a constant
+* `conformal_at_iff`: an equivalent definition of the conformality at some point
+* `conformal_at_preserves_angle`: if a map is `conformal_at` some point, then its differential
+                                  preserves all angles at that point
+* `conformal_groupoid`: the groupoid of conformal local homeomorphisms
+
+## Tags
+
+conformal
+
+## Warning
+
+The definition of conformality in this file does NOT require the maps to be orientation-preserving.
+Maps such as the conjugate function on the complex plane is considered as conformal maps.
+-/
+
+noncomputable theory
+
+section conformality
+
+/-- A continuous linear map `f'` is said to be conformal if it's
+a nonzero multiple of a bijective linear isometry. -/
+def is_conformal_map {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f' : X →L[ℝ] Y) :=
+∃ (c : ℝ) (hc : c ≠ 0) (lie : X ≃ₗᵢ[ℝ] Y), (f' : X → Y) = c • lie
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at {X Y : Type*} [inner_product_space ℝ X] [inner_product_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map f'
+
+
+namespace conformal_at
+
+variables {X Y Z : Type*}
+  [inner_product_space ℝ X] [inner_product_space ℝ Y] [inner_product_space ℝ Z]
+
+open linear_isometry_equiv continuous_linear_map
+
+theorem differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _, _, _, _⟩ := h in h₁.differentiable_at
+
+theorem id (x : X) : conformal_at id x :=
```

Comment:
To be parallel with (eg) the `differentiable_at` theory, this should move outside the `conformal_at` namespace, so that its full name is `conformal_at_id` rather than `conformal_at.id`.  Same for `conformal_on` and `conformal`.